src/anomalib/pre_processing/pre_processing.py

```diff
@@ -0,0 +1,177 @@
+        4. Default transforms (lowest priority)
+
+    Examples:
+        >>> from torchvision.transforms.v2 import Compose, Resize, ToTensor
+        >>> from anomalib.pre_processing import PreProcessor
+
+        >>> # Define transforms
+        >>> train_transform = Compose([Resize((224, 224)), ToTensor()])
+        >>> val_transform = Compose([Resize((256, 256)), CenterCrop((224, 224)), ToTensor()])
+
+        >>> # Create PreProcessor with stage-specific transforms
+        >>> pre_processor = PreProcessor(
+        ...     train_transform=train_transform,
+        ...     val_transform=val_transform
+        ... )
+
+        >>> # Create PreProcessor with a single transform for all stages
+        >>> common_transform = Compose([Resize((224, 224)), ToTensor()])
+        >>> pre_processor_common = PreProcessor(transform=common_transform)
+
+        >>> # Use in a Lightning module
+        >>> class MyModel(LightningModule):
+        ...     def __init__(self):
+        ...         super().__init__()
+        ...         self.pre_processor = PreProcessor(...)
+        ...
+        ...     def configure_callbacks(self):
+        ...         return [self.pre_processor]
+        ...
+        ...     def training_step(self, batch, batch_idx):
+        ...         # The pre_processor will automatically apply the correct transform
+        ...         processed_batch = self.pre_processor(batch)
+        ...         # Rest of the training step
+    """
+
+    def __init__(
+        self,
+        train_transform: Transform | None = None,
+        val_transform: Transform | None = None,
+        test_transform: Transform | None = None,
+        transform: Transform | None = None,
+    ) -> None:
+        super().__init__()
+
+        if transform and any([train_transform, val_transform, test_transform]):
+            msg = (
+                "`transforms` cannot be used together with `train_transform`, `val_transform`, `test_transform`.\n"
+                "If you want to apply the same transform to the training, validation and test data, "
+                "use only `transforms`. \n"
+                "Otherwise, specify transforms for training, validation and test individually."
+            )
+            raise ValueError(msg)
+
+        self.train_transform = train_transform or transform
+        self.val_transform = val_transform or transform
+        self.test_transform = test_transform or transform
+        self.predict_transform = self.test_transform
+        self.export_transform = get_exportable_transform(self.test_transform)
+
+    def setup_datamodule_transforms(self, datamodule: "AnomalibDataModule") -> None:
+        """Set up datamodule transforms."""
+        # If PreProcessor has transforms, propagate them to datamodule
+        if any([self.train_transform, self.val_transform, self.test_transform]):
+            transforms = {
+                "train": self.train_transform,
```

Comment:
This should be `fit` to be in line with the terminology used [here](https://github.com/samet-akcay/anomalib/blob/add-pre-processor/src/anomalib/pre_processing/utils/transform.py#L102) in the utils function.
```suggestion
                "fit": self.train_transform,
```